### ISSUE TITLE
CLI update check

### ISSUE THREAD

**Issue by brianteeman** [joomla/joomla-cms#40444]
### Steps to reproduce the issue
On a site that needs updating
Make sure that your database structure is reporting an issue
eg edit the value of version_id in the #_schemas table

Try to update with the update component
You can not because of the database structure error which is a required check

Now try to update using the cli

`php cli/joomla.php core:update

### Expected result
Update is not able to proceed because of the database structure issue

### Actual result
Update takes place

### Additional comments
Either the database structure check test is not important and shouldnt be required in the component OR the cli update should check the structure befiore installing.

I suspect the correct thing is the latter

**Comment by richard67** [comment]
I would say this applies to all required checks in the component.

**Comment by alikon** [comment]
what exactly do you mean ?
all that are present here https://github.com/joomla/joomla-cms/blob/c93bdfdbec16cac5e4ff1f774db29f8caab3dcd7/administrator/components/com_joomlaupdate/src/Model/UpdateModel.php#L1027

btw for the schema see #40461

**Comment by richard67** [comment]
> what exactly do you mean ? all that are present here
> 
> https://github.com/joomla/joomla-cms/blob/c93bdfdbec16cac5e4ff1f774db29f8caab3dcd7/administrator/components/com_joomlaupdate/src/Model/UpdateModel.php#L1027

Yes.

> 
> btw for the schema see #40461

The schema check would also included in the above getPhpOptions method:

https://github.com/joomla/joomla-cms/blob/c93bdfdbec16cac5e4ff1f774db29f8caab3dcd7/administrator/components/com_joomlaupdate/src/Model/UpdateModel.php#L1099-L1104

**Comment by alikon** [comment]
that's pretty unusable from CLI, it was introduced in 3.10 iirc, so...

**Comment by richard67** [comment]
@alikon P.S.: Your PR #40444 just adds a check for the database schema version. But what the check in the component does with the `$this->getDatabaseSchemaCheck()` is that it runs the complete database check like the database checker does when you go to it. I.e. it does not only check for that error with not matching schema version but for any structure errors. The not matching schema version in Brian's issue was just an example how to make that test fail. So I think your PR #40444 is not really sufficient.

**Comment by alikon** [comment]
yes i know that, and it will imply a complete  re-think of the whole "thing" on both sides

...1 small step at time...

**Comment by richard67** [comment]
Well, but the PR does not completely solve the issue, it solves only a part, so that should be made clear in the description of that PR.

**Comment by alikon** [comment]
that's why i've not closed this ... i'll update the pr description too for avoid confusion  👍
